### PR TITLE
Fix incremental build of tasks.proj for mobile

### PR DESCRIPTION
**Pull request by akoeplinger**
The semaphore file that is used as the input for deciding whether to rebuild tasks.proj wasn't properly taking the mobile task projects into account.
This resulted in e.g. WasmAppBuilder not being built if you built for desktop before, resulting in a build error.

We now use the same conditions for the semaphore file as we do for the project references.

### CODE REVIEW

File: tools-local/tasks/tasks.proj [hunk 1]
```diff
@@ -47,6 +47,16 @@
...
+      <TasksSrc Remove="$(MSBuildThisFileDirectory)mobile.tasks\AndroidAppBuilder\AndroidAppBuilder.csproj"
+                        Condition="'$(TargetOS)' != 'Android'" />
+      <TasksSrc Remove="$(MSBuildThisFileDirectory)mobile.tasks\AppleAppBuilder\AppleAppBuilder.csproj"
+                        Condition="'$(TargetOS)' != 'iOS' and '$(TargetOS)' != 'tvOS'" />
+      <TasksSrc Remove="$(MSBuildThisFileDirectory)mobile.tasks\WasmAppBuilder\WasmAppBuilder.csproj"
+                        Condition="'$(TargetOS)' != 'Browser'" />
+      <TasksSrc Remove="$(MSBuildThisFileDirectory)mobile.tasks\WasmBuildTasks\WasmBuildTasks.csproj"
+                        Condition="'$(TargetOS)' != 'Browser'" />
+      <TasksSrc Remove="$(MSBuildThisFileDirectory)mobile.tasks\AotCompilerTask\MonoAOTCompiler.csproj"
+                        Condition="'$(TargetsMobile)' != 'true'" />
```
Comment (ViktorHofer) [inline tools-local/tasks/tasks.proj:59]:
```suggestion
      <TasksSrc Include="@(ProjectReference)" />
```

Comment (akoeplinger) [reply tools-local/tasks/tasks.proj]: That makes much more sense indeed, thanks :)

**Comment by akoeplinger** [comment]
@ViktorHofer thinking about it, do we even need the semaphore file? can we just pass `@(ProjectReference)` as the Inputs for the BuildAndRestoreIncrementally target?

File: tools-local/tasks/tasks.proj [hunk 2]
```diff
@@ -46,7 +46,7 @@
     </PropertyGroup>
 
     <ItemGroup>
-      <TasksSrc Include="$(MSBuildThisFileDirectory)**\*.cs*;$(MSBuildThisFileDirectory)**\*.*proj" />
+      <TasksSrc Include="@(ProjectReference)" />
```
Comment (ViktorHofer) [inline tools-local/tasks/tasks.proj:49]: Hmm, `$(MSBuildThisFileDirectory)**\*.cs*;` is now missing. Give a minute and I will come up with a solution for it.

Comment (akoeplinger) [reply tools-local/tasks/tasks.proj:49]: does it matter? that should be handled by the *proj include of the ProjectReference?

**Comment by ViktorHofer** [comment]
> @ViktorHofer thinking about it, do we even need the semaphore file? can we just pass @(ProjectReference) as the Inputs for the BuildAndRestoreIncrementally target?

The semaphore file is the target's output, not input. It is still needed for the up-to-date file checks to work.

File: tools-local/tasks/tasks.proj (hunk 2 above)
Comment (ViktorHofer) [reply tools-local/tasks/tasks.proj:49]: It matters. MSBuild checks the timestamp of the project files but not of their source code. It doesn't interpret the project files at all for the up-to-date check.

**Review by ViktorHofer — APPROVED** [review]

**Review by safern — COMMENTED** [review]
LGTM

**Pull request merged**